core/vm/interpreter.go

```diff
@@ -940,6 +940,7 @@ func (in *GethEVMInterpreter) Step(state *GethState) bool {
 		state.logged = true
 	}
 
+	in.returnData = state.Result
```

Comment:
I am not against it, I just think it is not really the right place for this to be done here. This assignment is done while executing every single instruction using the geth EVM, while it should only be done once.

A solution could be to introduce another entry point for the CT framework, something like
```
func (in *GethEVMInterpreter) StepN(state *GethState) bool {
    in.returnData = state.Result
    // <space for other StepN preparation code>
    for i := 0; i < numSteps && !state.Halted; i++ {
	if !in.Step(state) {
		return false
	}
   }
   return true
}
```
and then call this one from Tosca's CT framework instead of having this look in Tosca.
This would introduce a location for performing setup tasks for `StepN` calls that are made from the `runPlain` function above. The `Step(..)` function could then also made private.

However, this might be left for a future task. For now I would suggest this:
```suggestion
	// This copies the result state from an input state into the internal state. It is needed
	// when running conformance tests to get the setup done. It would only be needed once,
	// but this requires additional refactoring. TODO: fix this!
	// When running regular contract executions, both fields are always in sync. Thus, this
	// is effectively a no-op.
	in.returnData = state.Result
```